apache/netbeans
# Remove downloader from NBI

- the download manager relied on suspend/resume/stop from java.lang.Thread which were deprecated since Java 1.2
 - JDK 23 removes suspend/resume, others are already no-ops
 - fixing this would require a completely different approach, and since NBI is no longer maintained and in the process to be replaced by nbpackage this is the easiest option

how to test:
1) setup JDK 23
2) checkout this branch
3) build (with zip):
```
ant clean -q
ant build -Dcluster.config=release
```
4) build installers (download [installer.sh](https://github.com/apache/netbeans-jenkins-lib/blob/master/resources/org/apache/netbeans/installer.sh) and run using bash)
```
bash installer.sh nbbuild/NetBeans-dev-dev-baa644f9077e618cb3555695e0d3f0a07a0db380-release.zip 42 121212
```

this should create the windows installer:
```
tree dist/
dist/
├── bundles
│   ├── Apache-NetBeans-42-bin-linux-x64.sh
│   └── Apache-NetBeans-42-bin-windows-x64.exe
```
which is now testable on windows (not testable on JDK 23ea since the native launcher can't parse ea java versions)

this PR would unblock https://github.com/apache/netbeans/pull/7525 (and https://github.com/apache/netbeans/pull/7484)
full test on JDK 23 and nb-javac 23 ran in https://github.com/apache/netbeans/pull/7538
related to https://github.com/apache/netbeans/issues/4952
